### PR TITLE
Bugfix/3943 docker container improvements

### PR DESCRIPTION
**Pull request by Callum17**
## Checklist:

- [✓] I have read the [Contributing guidelines](https://github.com/oobabooga/text-generation-webui/wiki/Contributing-guidelines).

- less verbose docker compose
- drops containerized virtualenv in favor of non root runtime user
- slightly smaller runtime image
- moved all dependencies to builder pattern - so it should be easier to build and test newer dependencies

Refactored this PR last moment as the original exllamav2 3943 issue was fixed here:
https://github.com/oobabooga/text-generation-webui/pull/4095

We should be able to close this issue:
https://github.com/oobabooga/text-generation-webui/issues/3943

My only question might be around the bump from flask_cloudflared==0.0.12 to flask_cloudflared==0.0.14 in the openai extension. Had to be bumped to prevent conflicts with the api extension. I still need to test this change. Functionally should be no different from the version installed by the current docker compose setup - unlike the current setup this builder pattern will explicitly fail if any constraints are violated in any of the requirements files.

Also worth noting that a number of conflicts were found against the (outdated?) requirements.txt in oobabooga/GPTQ-for-LLaMa. Conclusion was that it can be ignored.

inference successfully tested with:
ctransformers
exllama
exllamav2
llama.cpp
AutoGPTQ

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,10 @@ models/config-user.yaml
 Thumbs.db
 .chroma
 installer_files
+
+# ignore custom container envrionment
```
Comment (LoopControl) [inline .gitignore:39]:
```suggestion
# ignore custom container environment
```

**Comment by oobabooga** [comment]
Could you merge the dev branch and check if the updated Dockerfile still works? 

A relevant detail is that the OpenAI API is now the default, and it listens on port 5000 only (5005 is not used anymore). 

I don't know how to use Docker properly, but if you find that your updates still work, I can merge the PR based on your feedback.

**Comment by Callum17** [comment]
I've got an updated copy working against current HEAD of main branch. Still needs a minor tidy and the dev merge, but it's pretty much there.

will look to update this in the next day of two when I get time.

**Comment by oobabooga** [comment]
Awesome @Callum17, thank you.

**Comment by Callum17** [comment]
Think GitHub auto closed this when I was aligning with the dev branch.

Opened a clean PR here:
https://github.com/oobabooga/text-generation-webui/pull/4768

There are some issues (not created by me!) - most notably with GPTQ-for-Llama conflicting with the core package dependencies. I suspect we will start seeing issues in non docker installs if these aren't resolved.

**Pull request closed without merging**